### PR TITLE
refactor: connector moves and renames

### LINKED ISSUE #96: Connector moves and renames
Renames to allow for other endpoints in future (we'll get the cloudant part from the package name):
* `CloudantSourceConnector` -> `ChangesSourceConnector`
* `CloudantSinkConnector` -> `ChangesSinkConnector`

Put the "public" content in these packages
`com.ibm.cloud.cloudant.kafka.connect.connectors` (for the *SourceConnector and *SinkConnector classes)
`com.ibm.cloud.cloudant.kafka.connect.transforms` (for the transforms)
`com.ibm.cloud.cloudant.kafka.connect.transforms.predicates` (for the predicates)

Put the other content in a sensible set of sub-packages in 
`com.ibm.cloud.cloudant.kafka.connect.internal`

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Related to #96

## Approach

Renames under the `src/main/java/com/ibm/cloud/cloudant/kafka/` path:
```
connect/CloudantSinkConnector.java -> SinkConnector.java
connect/CloudantSourceConnector.java -> SourceChangesConnector.java
connect/CloudantConfigValidator.java -> connect/ConfigValidator.java
connect/CloudantConnectorConfig.java -> connect/ConnectorConfig.java
connect/CloudantSinkConnectorConfig.java -> connect/SinkConnectorConfig.java
connect/CloudantSinkTask.java -> connect/SinkTask.java
connect/CloudantSinkTaskConfig.java -> connect/SinkTaskConfig.java
connect/CloudantSourceConnectorConfig.java -> connect/SourceChangesConnectorConfig.java
connect/CloudantSourceTask.java -> connect/SourceChangesTask.java
connect/CloudantSourceTaskConfig.java -> connect/SourceChangesTaskConfig.java
transforms/ArrayFlatten.java -> connect/transforms/ArrayFlatten.java
transforms/MapToStruct.java -> connect/transforms/MapToStruct.java
transforms/predicates/IsDesignDocument.java -> connect/transforms/predicates/IsDesignDocument.java
```

Renames under the `src/test/java/com/ibm/cloud/cloudant/kafka/` path:
```
connect/CloudantSinkConnectorTest.java -> SinkConnectorTest.java
connect/CloudantSourceConnectorTest.java -> SourceChangesConnectorTest.java
connect/CloudantSinkErrorTest.java -> connect/SinkErrorTest.java
connect/CloudantSinkTaskTest.java -> connect/SinkTaskTest.java
connect/CloudantSourceAndSinkTest.java -> connect/SourceChangesAndCloudantSinkTest.java
connect/CloudantSourceTaskTest.java -> connect/SourceChangesTaskTest.java
transforms/ArrayFlattenTest.java -> connect/transforms/ArrayFlattenTest.java
transforms/MapToStructTest.java -> connect/transforms/MapToStructTest.java
transforms/predicates/IsDesignDocumentTest.java -> connect/transforms/predicates/IsDesignDocumentTest.java
performance/CloudantSinkPerformanceTest.java -> performance/SinkPerformanceTest.java
performance/CloudantSourceAndSinkPerformanceTest.java -> performance/SourceChangesAndSinkPerformanceTest.java
performance/CloudantSourcePerformanceTest.java -> performance/SourceChangesPerformanceTest.java
```

I also refactored the references in other files to reflect the new class/file/package names.

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A file rename/move changes

## Monitoring and Logging
<!--
EITHER:

The log properties changed accordingly.

### CODE REVIEW

File: src/main/java/com/ibm/cloud/cloudant/kafka/connect/connectors/ChangesSourceConnector.java [hunk 1]
```diff
@@ -11,10 +11,14 @@
  * either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
-package com.ibm.cloud.cloudant.kafka.connect;
+package com.ibm.cloud.cloudant.kafka.connect.connectors;
```
Comment (ricellis) [inline src/main/java/com/ibm/cloud/cloudant/kafka/connect/connectors/ChangesSourceConnector.java:14]: Looking at this again it does seem a bit over-long especially as it needs entering into config.
Maybe we should drop one of hte levels here.

**Review by ricellis — COMMENTED** [review]
I think this is ok, but we will need a follow-up PR to move some of the other classes into more sensible packages too.
A couple of suggestions from me, but I think Tom has some similar thoughts and will probably give more detail.

**Comment by tomblench** [comment]
Looks good, but I would suggest two tweaks:
- Move the two connector classes up to `com.ibm.cloud.cloudant.kafka`: these are user-facing names which users will have to type, so I think they should be as short as feasibly possible
- `CloudantSinkConnector` should just become `SinkConnector` to more closely mirror the source connector
- The source connector should become `SourceChangesConnector` so that there is consistency with the sink connector, where the first word is source or sink.

In summary:
`com.ibm.cloud.cloudant.kafka.SourceChangesConnector`
`com.ibm.cloud.cloudant.kafka.SinkConnector`

File: src/main/java/com/ibm/cloud/cloudant/kafka/connect/CloudantSinkConnectorConfig.java [hunk 2]
```diff
@@ -20,12 +20,12 @@
 
 import java.util.Map;
 
-public class CloudantSinkConnectorConfig extends CloudantConnectorConfig {
+public class CloudantSinkConnectorConfig extends ConnectorConfig {
```
[the remark this answers is not in the document]
Comment (mojito317) [reply src/main/java/com/ibm/cloud/cloudant/kafka/connect/CloudantSinkConnectorConfig.java:23]: You mean renaming `CloudantSinkConnector` to `SinkConnector` too? The problem is there's a kafka class (that this sink connector implements) named `SinkConnector` too: `public class SinkConnector extends org.apache.kafka.connect.sink.SinkConnector`

File: src/main/java/com/ibm/cloud/cloudant/kafka/connect/connectors/ChangesSourceConnector.java (hunk 1 above)
Comment (mojito317) [reply src/main/java/com/ibm/cloud/cloudant/kafka/connect/connectors/ChangesSourceConnector.java:14]: As @tomblench suggested, I will put them on the level `com.ibm.cloud.cloudant.kafka`

Comment (mojito317) [reply src/main/java/com/ibm/cloud/cloudant/kafka/connect/connectors/ChangesSourceConnector.java]: It is fixed in [8bf0292](https://github.com/IBM/cloudant-kafka-connector/pull/110/commits/8bf0292e64bf6eb92a3a7a33f9e97bc90a65e83d)

**Comment by mojito317** [comment]
> * Move the two connector classes up to `com.ibm.cloud.cloudant.kafka`: these are user-facing names which users will have to type, so I think they should be as short as feasibly possible

This is done in [8bf0292](https://github.com/IBM/cloudant-kafka-connector/pull/110/commits/8bf0292e64bf6eb92a3a7a33f9e97bc90a65e83d) and tests are moved up in [59772b2](https://github.com/IBM/cloudant-kafka-connector/pull/110/commits/59772b2e1cc7bf8c9c1f6b02a98ccd0be35a3bd7)

> * `CloudantSinkConnector` should just become `SinkConnector` to more closely mirror the source connector

I have not touched this yet, because of an ongoing conversation here: https://github.com/IBM/cloudant-kafka-connector/pull/110#discussion_r981035586

> * The source connector should become `SourceChangesConnector` so that there is consistency with the sink connector, where the first word is source or sink.

This is fixed in [59772b2](https://github.com/IBM/cloudant-kafka-connector/pull/110/commits/59772b2e1cc7bf8c9c1f6b02a98ccd0be35a3bd7)

File: src/main/java/com/ibm/cloud/cloudant/kafka/connect/CloudantSinkConnectorConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomblench) [reply src/main/java/com/ibm/cloud/cloudant/kafka/connect/CloudantSinkConnectorConfig.java:23]: Good question - personally I don't mind the name clash. Users will generally only ever use this name when they configure the connector, where they have to give the fully qualified name, eg
`connector.class=com.ibm.cloud.cloudant.kafka.SinkConnector`

[the remark this answers is not in the document]
Comment (mojito317) [reply src/main/java/com/ibm/cloud/cloudant/kafka/connect/CloudantSinkConnectorConfig.java]: Ready in https://github.com/IBM/cloudant-kafka-connector/pull/110/commits/3c7967185013bbf8f880c99e2e0333a93630d884

**Review by tomblench — APPROVED** [review]

**Review by ricellis — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews @tomblench and @ricellis! I've also added a line to the `CHANGES.md` file, so now I'm ready to merge this PR.
